controllers/space/space_controller_test.go

```diff
@@ -1510,6 +1511,46 @@ func TestUpdateSpaceTier(t *testing.T) {
 			AssertThatCountersAndMetrics(t).
 				HaveSpacesForCluster("member-1", 1) // space counter is unchanged
 		})
+
+		t.Run("the templateref is not yet updated in status.revisions", func(t *testing.T) {
+			// given that Space is promoted from `base1ns` to `other` tier and corresponding NSTemplateSet is not up-to-date
+			otherTier := tiertest.OtherTier()
+			s := spacetest.NewSpace(test.HostOperatorNs, "oddity",
+				spacetest.WithTierName(otherTier.Name),      // tier changed to other tier
+				spacetest.WithTierHashLabelFor(base1nsTier), // still has the old tier hash label
+				spacetest.WithSpecTargetCluster("member-1"),
+				spacetest.WithStatusTargetCluster("member-1"), // already provisioned on a target cluster
+				spacetest.WithFinalizer(),
+				spacetest.WithCondition(spacetest.Ready()))
+			hostClient := test.NewFakeClient(t, s, base1nsTier, otherTier)
+			nstmplSet := nstemplatetsettest.NewNSTemplateSet("oddity", nstemplatetsettest.WithReferencesFor(base1nsTier), nstemplatetsettest.WithReadyCondition())
+			member1Client := test.NewFakeClient(t, nstmplSet)
+			member1 := NewMemberClusterWithClient(member1Client, "member-1", corev1.ConditionTrue)
+			member2 := NewMemberClusterWithTenantRole(t, "member-2", corev1.ConditionTrue)
+			ctrl := newReconciler(hostClient, member1, member2)
+			ctrl.LastExecutedUpdate = time.Now().Add(-1 * time.Minute) // assume that last executed update happened a long time ago
+			InitializeCounters(t,
+				NewToolchainStatus(
+					WithMetric(toolchainv1alpha1.UserSignupsPerActivationAndDomainMetricKey, toolchainv1alpha1.Metric{
+						"1,internal": 1,
+					}),
+					WithMetric(toolchainv1alpha1.MasterUserRecordsPerDomainMetricKey, toolchainv1alpha1.Metric{
+						string(metrics.Internal): 1,
+					}),
+					WithMember("member-1", WithSpaceCount(1)),
+				))
```

Comment:
no need to test anything with the counters
```suggestion
			InitializeCounters(t, NewToolchainStatus())
```